internal/security/bootstrapper/mosquitto/handlers/handlers.go

```diff
@@ -0,0 +1,169 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software distributed under the License
+* is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+* or implied. See the License for the specific language governing permissions and limitations under
+* the License.
+*
+*******************************************************************************/
+
+package handlers
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"html/template"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+
+	"github.com/edgexfoundry/edgex-go/internal/security/bootstrapper/mosquitto/container"
+
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/secret"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/startup"
+	bootstrapConfig "github.com/edgexfoundry/go-mod-bootstrap/v2/config"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/di"
+)
+
+const (
+	// file read/write permission only for owner
+	readWriteOnlyForOwner os.FileMode = 0600
+)
+
+// Handler is the redis bootstrapping handler
+type Handler struct {
+	credentials bootstrapConfig.Credentials
+}
+
+// NewHandler instantiates a new Handler
+func NewHandler() *Handler {
+	return &Handler{}
+}
+
+// GetCredentials retrieves the mosquitto credentials from secretstore
+func (handler *Handler) GetCredentials(ctx context.Context, _ *sync.WaitGroup, startupTimer startup.Timer,
+	dic *di.Container) bool {
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	config := container.ConfigurationFrom(dic.Get)
+	secretProvider := bootstrapContainer.SecretProviderFrom(dic.Get)
+
+	var credentials *bootstrapConfig.Credentials
+
+	for startupTimer.HasNotElapsed() {
+		// retrieve mosquitto credentials from secretstore
+		secrets, err := secretProvider.GetSecret(config.MessageQueue.SecretName)
```

Comment:
Humm, `config.MessageQueue.SecretName` would need to be passed in some how to keep it generic for use for Redis DB creds. This is the only difference between the handle for RedisDB, correct?